lighthouse-core/gather/connections/extension.js

```diff
@@ -112,19 +112,21 @@ class ExtensionConnection extends Connection {
 
       chrome.debugger.sendCommand({tabId: this._tabId}, command, params, result => {
         if (chrome.runtime.lastError) {
+          // The error from the extension has a `message` property that is the
+          // stringified version of the actual protocol error object.
           const message = chrome.runtime.lastError.message;
           let error;
           try {
             error = JSON.parse(message);
           } catch (e) {}
           error = error || {message: 'Unknown debugger protocol error.'};
 
-          const callback = {
-            resolve,
-            reject,
-            method: command
-          };
-          return this.handleRawError(error, callback);
+          // handleRawError returns or throws synchronously, so try/catch awkwardly.
+          try {
+            return resolve(this.handleRawError(error, command));
+          } catch (err) {
+            return reject(err);
+          }
         }
```

Comment:
This pattern is a little awkward (as noted) but I think I prefer this flavor to what we had here before.